### ISSUE TITLE
Custom tileset rendering error

### ISSUE THREAD

**Issue by PatrickZenker** [mapbox/mapbox-gl-native#8678]
Hey,

i've got a rendering error on several zoom levels with the latest SDK version (5.0.2).
With SDK version 4.2.0 I had no problems and if I style my Mapbox map everything looks fine (like the first screenshot).

**Platform: Android**
**Mapbox SDK version: 5.0.2 **

### Steps to trigger behavior

 Use a custom tileset

### Expected behavior
Sharp maptiles on every zoom level like this:
![3](https://cloud.githubusercontent.com/assets/23097192/24792016/a1312d34-1b7c-11e7-986b-c9e26bbd328a.jpg)
### Actual behavior
Black rendering boxes? 
![2](https://cloud.githubusercontent.com/assets/23097192/24792017/a131dcb6-1b7c-11e7-8d98-8d0354adc23a.jpg)
![1](https://cloud.githubusercontent.com/assets/23097192/24792001/917c6dc2-1b7c-11e7-9203-b7161e20cfe5.jpg)

**Comment by jfirebaugh** [comment]
Can you provide the style with custom tileset you're using?

**Comment by PatrickZenker** [comment]
Style URL: mapbox://styles/patrickzenker/ciuton0ty00yv2hpbjgv5l4df
Lat/Lng: 53.874336,10.683844

**Comment by jfirebaugh** [comment]
Thank you -- I was able to easily reproduce the issue with your style. The cause of the aliasing seen around the edges of the opaque portions of the raster layer is that tiles from lower zoom levels are being rendered underneath the desired tiles. These low zoom tiles are overzoomed for rendering, producing undesired artifacts. You can see this more clearly when turning on tile borders:

![image](https://cloud.githubusercontent.com/assets/98601/24978378/506bfcb4-1f85-11e7-9a7f-fbd5ed9daa66.png)

These lower zoom tiles are loaded by an overly-aggressive tile loading algorithm, which is detecting that some tiles cannot be loaded at the desired zoom (because they are outside of the bounds of the raster source and don't exist), and loads lower zoom tiles in an effort to fully cover the visible area. This is a bug; we'll need to either adjust what tiles are loaded in this situation or ensure that we clip out child tiles when rendering the lower zoom tiles.

cc @kkaefer

**Comment by jfirebaugh** [comment]
Might be addressed by https://github.com/mapbox/mapbox-gl-native/issues/5394.

**Comment by kkaefer** [comment]
Fixed by #9468.

**Comment by P-Zenker** [comment]
@kkaefer @jfirebaugh 

With 
```
    compile('com.mapbox.mapboxsdk:mapbox-android-sdk:5.1.4@aar') {
        transitive = true
    }
    compile 'com.mapbox.mapboxsdk:mapbox-android-telemetry:2.2.5'
    compile 'com.mapbox.mapboxsdk:mapbox-android-services:2.2.5'
    compile('com.mapbox.mapboxsdk:mapbox-android-navigation-ui:0.7.0-SNAPSHOT') {
        transitive = true
    }
```
I still got the issue.

![bildschirmfoto 2017-10-06 um 18 26 33](https://user-images.githubusercontent.com/23097192/31288116-ea48aaac-aac3-11e7-8e5e-13f8295b2811.png)
![bildschirmfoto 2017-10-06 um 18 26 22](https://user-images.githubusercontent.com/23097192/31288117-ea69160c-aac3-11e7-9182-f8e5a5c68280.png)

Am I doing something wrong when I convert my images to MBTiles (to upload them into Mapbox Studio)?
